ui-src/src/app/configurations/configurations.component.ts

```diff
@@ -0,0 +1,155 @@
+      numIterations: { type: "integer", minimum: 1 },
+      concurrentUsers: { type: "integer", minimum: 1 },
+      allowablePeakMemoryVariance: {
+        type: "number",
+        minimum: 0,
+        maximum: 100
+      },
+      allowableServiceResponseTimeVariance: {
+        type: "number",
+        minimum: 0,
+        maximum: 100
+      },
+      testSuite: {
+        type: "string",
+        enum: ["Default-1", "Default-2", "Default-3"]
+      },
+      requestDelay: { type: "integer", minimum: 1 },
+      TPSFreq: { type: "integer", minimum: 1 },
+      rampUsers: { type: "integer", minimum: 1 },
+      rampDelay: { type: "integer", minimum: 0 },
+      testCaseDir: { type: "string" },
+      testSuiteDir: { type: "string" },
+      baseStatsOutputDir: { type: "string" },
+      reportOutputDir: { type: "string" }
+    },
+    required: [
+      "apiName",
+      "targetHost",
+      "targetPort",
+      "numIterations",
+      "concurrentUsers",
+      "allowablePeakMemoryVariance",
+      "allowableServiceResponseTimeVariance",
+      "testSuite",
+      "requestDelay",
+      "TPSFreq",
+      "rampUsers",
+      "rampDelay",
+      "testCaseDir",
+      "testSuiteDir",
+      "baseStatsOutputDir",
+      "reportOutputDir"
+    ]
+  };
+
+  exampleData = {
+    allowablePeakMemoryVariance: 15,
+    allowableServiceResponseTimeVariance: 15
+  };
+
+  layout = [
+    {
+      type: "flex",
+      "flex-flow": "row wrap",
+      items: [
+        "apiName",
+        "numIterations",
+        {
+          key: "requestDelay",
+          title: "Request Delay (ms)"
+        }
+      ]
+    },
+    {
+      type: "flex",
+      "flex-flow": "row wrap",
+      items: [
+        "targetHost",
+        "concurrentUsers",
+        {
+          key: "TPSFreq",
+          title: "TPS Frequency (s)"
+        }
+      ]
+    },
+    {
+      type: "flex",
+      "flex-flow": "row wrap",
+      items: [
+        "targetPort",
+        {
+          key: "allowablePeakMemoryVariance",
+          title: "Memory Variance (%)"
+        },
+        "rampUsers"
+      ]
+    },
+    {
+      type: "flex",
+      "flex-flow": "row wrap",
+      items: [
+        "memoryEndpoint",
+        {
+          key: "allowableServiceResponseTimeVariance",
+          title: "Service Variance (%)"
+        },
+        {
+          key: "rampDelay",
+          title: "Ramp Delay (s)"
+        }
+      ]
+    },
+    { key: "testSuite" },
+    { key: "testCaseDir", title: "Test Case Directory" },
+    { key: "testSuiteDir", title: "Test Suites Directory" },
+    { key: "baseStatsOutputDir", title: "Base Stats Output Directory" },
+    { key: "reportOutputDir", title: "Report Output Directory" }
+  ];
+
+  ngOnInit() {
+    this.formData = {
+      allowablePeakMemoryVariance: 15,
+      allowableServiceResponseTimeVariance: 15
+    };
+  }
+
+  onSubmit(configData) {
+    this.toastr.success("Your data has been save!", "Success!");
```

Comment:
this.automatedUIServices
     .createJsonFile(configData)
     .subscribe(data =>   this.toastr.success("Your data has been save!", "Success!"));